c2645637.lua

```diff
@@ -30,18 +30,21 @@ function c2645637.initial_effect(c)
 	c:RegisterEffect(e3)
 end
 function c2645637.disop(e,tp,eg,ep,ev,re,r,rp)
-	if Duel.GetChainInfo(ev,CHAININFO_TRIGGERING_LOCATION)==LOCATION_REMOVED and re:IsActiveType(TYPE_MONSTER) and rp==1-tp then
+	if rp==tp or e:GetHandler():IsStatus(STATUS_BATTLE_DESTROYED) then return end
```

Comment:
should be descripted in comment.

in fact, card with `STATUS_BATTLE_DESTROYED` cannot use this effect automatically. Have you tested this is really necessary?

Reply:
i'm sorry about changing it without testing.